### ISSUE TITLE
Update to Scalacheck 1.13

### ISSUE THREAD

**Issue by ceedubs** [scalatest/scalatest#837]
Scalacheck 1.13.0 is hot off the press and [includes handy things](https://github.com/rickynils/scalacheck/blob/1.13.0/RELEASE) like a fix for #661 and `Cogen` for generating "arbitrarier" functions. I'd like to see a version of scalatest that is compatible with this new Scalacheck, so then we can update the chain of [discipline](https://github.com/typelevel/discipline) and then [cats](http://typelevel.org/cats/). Does this sound like something that can be done in the near future? I'm happy to help with the effort, but I wanted to check in before starting.

**Comment by bvenners** [comment]
Yes,, we can make a release tomorrow. What version of scalatest are you currently using?

**Comment by ceedubs** [comment]
Thanks for the speedy turnaround @bvenners! It looks like [Cats is currently on 3.0.0-M7](https://github.com/typelevel/cats/blob/master/build.sbt#L163). I imagine we might as well update to the latest 3.0 as part of this effort, though.

**Comment by bvenners** [comment]
Sorry I am in Asia and the past day have had an intermittent internet connection. I am moving to a new place tonight and will try again from there.

**Comment by ceedubs** [comment]
No worries, @bvenners. In fact, you may want to hold off on this: https://github.com/rickynils/scalacheck/issues/218

**Comment by adelbertc** [comment]
Hey @bvenners ,

I'm working on [bumping the ScalaCheck dependency](https://github.com/typelevel/cats/pull/942) to 1.13.0 on Cats both to get better arbitrary FunctionN's, as well as to fix some build issues we're having (https://github.com/typelevel/cats/pull/938, https://github.com/typelevel/cats/issues/810), though running into issues due to the usual dependency bugs.

I'm thinking we may be missing a ScalaTest with ScalaCheck 1.13.0, and after I think we need to bump [Discipline](https://github.com/typelevel/discipline) to use the new ScalaTest and ScalaCheck.

Do the errors in https://travis-ci.org/typelevel/cats/jobs/117388024 look like issues that would result from these dependency issues?

Currently Cats is using ScalaTest 3.0.0-M7.

**Comment by bvenners** [comment]
DeferredAbortedSuite was our bug, which hides what the root cause is. We have fixed it since 3.0.0-M7. But IncompatibleClassChangeError hints at the ScalaCheck problem. Last I looked there was still a bug I thought ScalaCheck was going to fix before we could upgrade. I'll take a look to see if we can't upgrade now.

**Comment by adelbertc** [comment]
Awesome, thanks Bill!

**Comment by cheeseng** [comment]
PR is here: https://github.com/scalatest/scalatest/pull/861

**Comment by bvenners** [comment]
We released a 3.0.0-M16-SNAP2 release that has ScalaCheck 1.13.0 as the dependency.

https://oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.11/3.0.0-M16-SNAP2/

Please give it a try and let us know if that solves your issue.

Thanks.

Bill

**Comment by adelbertc** [comment]
Awesome, thanks @bvenners !

I tired doing a `publishLocal` on Discipline by bumping https://github.com/typelevel/discipline/blob/master/build.sbt#L26 but now getting the below error.. any ideas?

```
[error] missing or invalid dependency detected while loading class file 'Succeeded.class'.
[error] Could not access type Assertion in value org.scalatest.compatible,
[error] because it (or its dependencies) are missing. Check your build definition for
[error] missing or conflicting dependencies. (Re-run with `-Ylog-classpath` to see the problematic classpath.)
[error] A full rebuild may help if 'Succeeded.class' was compiled against an incompatible version of org.scalatest.compatible.
[info] No documentation generated with unsuccessful compiler run
[error] one error found
[error] missing or invalid dependency detected while loading class file 'Succeeded.class'.
[error] Could not access type Assertion in value org.scalatest.compatible,
[error] because it (or its dependencies) are missing. Check your build definition for
[error] missing or conflicting dependencies. (Re-run with `-Ylog-classpath` to see the problematic classpath.)
[error] A full rebuild may help if 'Succeeded.class' was compiled against an incompatible version of org.scalatest.compatible.
[error] one error found
```

**Comment by bvenners** [comment]
Oh, perhaps we didn't include the new package org.scalatest.compatible in the jar. That was probably added in since the previous SNAP release. We'll try again.

**Comment by adelbertc** [comment]
I think this is mostly messing with the JS build for Discipline

```
[error] missing or invalid dependency detected while loading class file 'Succeeded.class'.
[error] Could not access type Assertion in value org.scalatest.compatible,
[error] because it (or its dependencies) are missing. Check your build definition for
[error] missing or conflicting dependencies. (Re-run with `-Ylog-classpath` to see the problematic classpath.)
[error] A full rebuild may help if 'Succeeded.class' was compiled against an incompatible version of org.scalatest.compatible.
[info] No documentation generated with unsuccessful compiler run
[error] one error found
[error] missing or invalid dependency detected while loading class file 'Succeeded.class'.
[error] Could not access type Assertion in value org.scalatest.compatible,
[error] because it (or its dependencies) are missing. Check your build definition for
[error] missing or conflicting dependencies. (Re-run with `-Ylog-classpath` to see the problematic classpath.)
[error] A full rebuild may help if 'Succeeded.class' was compiled against an incompatible version of org.scalatest.compatible.
[error] one error found
[error] (disciplineJS/compile:doc) Scaladoc generation failed
[error] (disciplineJS/compile:compileIncremental) Compilation failed
```

**Comment by bvenners** [comment]
Hi Adelbert,

It was just a forgoto. I need to find a way to test the publish part of our build to make sure it includes new directories. This isn't the first SNAP to have something missing. Anyway, we have fixed it and redeployed here:

https://oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.11/3.0.0-M16-SNAP3/

Please give that a shot. Sorry to delegate our testing to you!

Bill

**Comment by adelbertc** [comment]
Hey Bill,

Look like that worked! I can `publishLocal` on `discipline` just fine now, as well as `publishLocal` on algebra.

Currently getting:

```
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.typelevel#discipline_2.11;0.5-SNAPSHOT: configuration not found in org.typelevel#discipline_2.11;0.5-SNAPSHOT: 'master'. It was required from org.spire-math#algebra-laws_2.11;0.3.2-SNAPSHOT compile
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn]
[warn] 	Note: Unresolved dependencies path:
[warn] 		org.typelevel:discipline_2.11:0.5-SNAPSHOT
[warn] 		  +- org.spire-math:algebra-laws_2.11:0.3.2-SNAPSHOT (<elided>/cats/build.sbt#L32)
[warn] 		  +- org.typelevel:cats-laws_2.11:0.5.0-SNAPSHOT ()
```

But I'm guessing that's not your issue :-)

Thanks for your help and quick response!

**Comment by adelbertc** [comment]
Hey Bill - to confirm only the 2.11 SNAPSHOT was published right? Looks like we're getting [errors](https://github.com/typelevel/discipline/pull/17) related to JS not being picked up, which is fine I think since we wouldn't want to depend on a SNAPSHOT anyways. We'll just wait for you to be comfortable with releasing a non-SNAPSHOT :-)

**Comment by bvenners** [comment]
No, they should all be there, including 2.10 and 2.11 for JVM and JS. For example, here's 2.11 JS:

https://oss.sonatype.org/content/groups/public/org/scalatest/scalatest_sjs0.6_2.11/3.0.0-M16-SNAP3/

This isn't a SNAPSHOT but a "SNAP" so that it is stable. It won't ever change.

**Comment by bvenners** [comment]
What it isn't is a Milestone release, because I haven't caught up with documentation. So it is not the worst idea to wait, but it may be a few weeks before M16. I actually hope to release 3.0 final for ScalaDays in early May, so once I get a few more things out of the way I'll be quite focused on ScalaTest/Scalactic until ScalaDays. Scala Scala Scala!

**Comment by adelbertc** [comment]
@bvenners - Ah cool, alright looks like it may be a different issue than. I'll check with the others to see if we're OK depending on a SNAP, if not we'll wait :-) Thanks for all your help!

**Comment by muuki88** [comment]
What is the current status?

**Comment by dmyersturnbull** [comment]
This should probably be for ScalaCheck 1.14.x now.

**Comment by jeffmay** [comment]
Unless we want to bump this to ScalaCheck 1.14.0, I believe this can be closed as ScalaCheck 1.13 is included as a dependency of ScalaTest 3.0.x
